autoblocks/_impl/testing/models.py

```diff
@@ -38,3 +38,6 @@ def id(self) -> str:
     @abc.abstractmethod
     def evaluate(self, test_case: BaseTestCase, output: Any) -> Evaluation:
         pass
+
+    def evaluateOnline(self, event: Any) -> Evaluation:
```

Comment:
not sure we want this on the same `BaseEvaluator` because we'll want to support all three of these:

* a user is only using offline evaluators
* a user is only using online evaluators
* a user is using both online and offline evaluators

currently this won't work for a user only using online evaluators because python will throw when they haven't implemented `evaluate`